lib/phoenix/transports/long_poll_server.ex

```diff
@@ -112,6 +112,11 @@ defmodule Phoenix.Transports.LongPoll.Server do
     end
   end
 
+  def handle_info({:graceful_exit, channel_pid, %Phoenix.Socket.Message{} = msg}, state) do
+    new_state = delete(state, msg.topic, channel_pid)
```

Comment:
OH. If we are deleting the entry from the state then we no longer need to unlink in the channel/server above. Can you please remove the unlink? Other than that this is good to go!